src/wp-includes/load.php

```diff
@@ -335,6 +337,20 @@ function wp_in_development_mode( $mode ) {
 		return true;
 	}
 
+	$valid_modes = array(
+		'core',
+		'plugin',
+		'theme',
+		'all',
+	);
+
+	/**
+	 * If no mode is passed, return true if any valid development mode is set.
+	 */
```

Comment:
CS Nit: single line comments use `//` (also `/**` is for docblocks rather than a multi-line comments)
```suggestion
	// If no mode is passed, return true if any valid development mode is set.
```